pennylane_cirq/qsim_device.py

```diff
@@ -53,13 +54,44 @@ class QSimDevice(SimulatorDevice):
 
     def __init__(self, wires, shots=1000, analytic=True, qubits=None):
         super().__init__(wires, shots, analytic, qubits)
-        self.circuit = None
+        self.circuit = qsimcirq.QSimCircuit(cirq_circuit=cirq.Circuit())
         self._simulator = qsimcirq.QSimSimulator()
 
     def reset(self):
         # pylint: disable=missing-function-docstring
         super().reset()
-        self.circuit = qsimcirq.QSimCircuit(cirq_circuit=self.circuit)
+        self.circuit = qsimcirq.QSimCircuit(cirq_circuit=cirq.Circuit())
+
+    def _apply_basis_state(self, basis_state_operation):
+        # pylint: disable=missing-function-docstring
+        if not self.analytic:
+            raise qml.DeviceError("The operation BasisState is only supported in analytic mode.")
```

Comment:
I'm curious why it doesn't work in non-analytic mode?

Reply:
Not sure actually. I copied it over from the SimlatorDevice class. :thinking:

Reply:
(Is this true for qsim? or is it a Cirq restriction?)

Reply:
It might be worth removing `BasisState` from the supported operations, so that PL will always default to the decomposition (e.g., PauliX gates), which should always be supported.